### PR TITLE
npm: dejavu-data-browser

### PR DESCRIPTION
**Pull request by ShahAnuj2610**

### CODE REVIEW

**Comment by ShahAnuj2610** [comment]
dependent on: https://github.com/appbaseio/batteries/pull/173

File: packages/browser/package.json [hunk 1]
```diff
@@ -13,35 +13,24 @@
...
-	},
-	"dependencies": {
+		"react-dom": "^16.5.1",
 		"@appbaseio/designkit": "^0.9.1",
 		"@appbaseio/reactivesearch": "^3.0.2",
-		"@divyanshu013/media": "^1.0.0",
 		"antd": "^3.10.4",
 		"appbase-js": "^4.0.2-beta.2",
 		"brace": "^0.11.1",
 		"codesandbox": "1.3.5",
```
Comment (lakhansamani) [inline packages/browser/package.json]: I don't think we need `codesandbox` or any packages related to Search Preview here, they should be part of `dejavu-main`.

You can try removing packages listed here and not used by `browser`
https://github.com/appbaseio/batteries

Same applies for dependencies.

**Review by lakhansamani — CHANGES_REQUESTED** [review]
Remove battiers dependencies which are not required by `dejavu-browser`

**Comment by lakhansamani** [comment]
@ShahAnuj2610 can you share build command + dist repositories that need to be set here?

File: packages/browser/package.json [hunk 2]
```diff
@@ -0,0 +1,64 @@
+{
+	"name": "dejavu-data-browser",
```
Comment (lakhansamani) [inline packages/browser/package.json]: @ShahAnuj2610 can you change this to `@appbaseio/dejavu-browser`

File: .eslintrc.js [hunk 3]
```diff
@@ -1,27 +1,27 @@
...
-		'react/jsx-filename-extension': [1, { extensions: ['.js'] }],
-		'react/forbid-prop-types': 0,
-		'react/destructuring-assignment': 0,
-		'react/require-default-props': 0,
-		'prettier/prettier': 'error',
-		'no-underscore-dangle': 0,
-		'jsx-a11y/click-events-have-key-events': 0,
-		'jsx-a11y/no-static-element-interactions': 0,
-	},
+  parser: 'babel-eslint',
```
Comment (lakhansamani) [inline .eslintrc.js]: Can you fix your editor config and use tab size as per editor config.

File: package.json [hunk 4]
```diff
@@ -99,35 +41,30 @@
...
-		]
+	"license": "MIT",
+	"scripts": {
+		"dev:browser": "lerna run watch --scope dejavu-data-browser --stream",
+		"dev:dejavu": "lerna run watch --scope dejavu-main --stream",
+		"dev:dejavu:site": "lerna run watch:site --scope dejavu-main --stream",
+		"build:browser": "lerna run build --scope dejavu-data-browser --stream",
+		"build:dejavu": "yarn build:browser && lerna run build --scope dejavu-main --stream",
+		"build:dejavu:app": "yarn build:browser && lerna run build:app --scope dejavu-main --stream",
+		"build:dejavu:site": "yarn build:browser && lerna run build:site --scope dejavu-main --stream"
```
Comment (lakhansamani) [inline package.json]: Can you add code formating using `prettier`

https://github.com/appbaseio/dejavu/blob/dev/package.json#L23

Also, chrome extension build is missing
https://github.com/appbaseio/dejavu/blob/dev/package.json#L24

Make sure extension & its build works correctly

File: packages/browser/src/components/ConnectApp/ConnectApp.js [hunk 5]
```diff
@@ -109,7 +112,7 @@ class ConnectApp extends Component<Props, State> {
 		// sync state from url
 		let appname = '';
 		let url = '';
-		const { mode, isConnected, isHidden } = this.props;
+		const { mode, isConnected, isHidden, forceReconnect } = this.props;
```
Comment (lakhansamani) [inline packages/browser/src/components/ConnectApp/ConnectApp.js:115]: Can you add a comment on when `foreceReconnect` is required?

**Review by lakhansamani — CHANGES_REQUESTED** [review]

**Comment by lakhansamani** [comment]
@ShahAnuj2610 can you bring docker file at the root
https://github.com/ShahAnuj2610/dejavu/blob/feat/browser-package/packages/dejavu-main/Dockerfile

And make changes for the folder structure there

**Comment by lakhansamani** [comment]
@ShahAnuj2610 there is one css bug
When you navigate to `search preview` and come back to `browser screen` 

observe horizontal scroll

File: packages/dejavu-main/app/src/components/OldDejavuBanner/OldDejavuBanner.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+const LOCAL_STORAGE_ITEM = 'showOldBanner';
+
+type Props = {
+  isConnected: boolean,
+  rawUrl: string,
+  appname: string,
+};
+
+const OldDejavuBanner = ({ appname, rawUrl, isConnected }: Props) => {
+	console.log('hello');
```
Comment (lakhansamani) [inline packages/dejavu-main/app/src/components/OldDejavuBanner/OldDejavuBanner.js]: Please remove log

**Comment by ShahAnuj2610** [comment]
@lakhansamani have done the changes. Here are some command line instructions

### How to publish to npm?

from the root of the project run the following command
```
yarn && yarn build:browser && cd packages/browser && npm publish
```

### How to run locally with watchers?

from the root of the project run the following command
```
yarn && yarn dev:browser && yarn dev:dejavu
```

### netlify config

**Pull request merged**